content/faq/errors.md

```diff
@@ -17,17 +17,49 @@ Potential solutions:
 
-If the `unknown_token` above is the string `dependency`, you might have a circular file import. This is different from the [circular dependency](./errors.md#circular-dependency-error) below because instead of having providers depend on each other in their constructors, it just means that two files end up importing each other. A common case would be a module file declaring a token and importing a provider, and the provider import the token constant from the module file. If you are using barrel files, ensure that your barrel imports do not end up creating these circular imports as well.
+If the `<unknown_token>` above is the string `dependency`, you might have a circular file import. This is different from the [circular dependency](/faq/common-errors#circular-dependency-error) below because instead of having providers depend on each other in their constructors, it just means that two files end up importing each other. A common case would be a module file declaring a token and importing a provider, and the provider import the token constant from the module file. If you are using barrel files, ensure that your barrel imports do not end up creating these circular imports as well.
+
+If the `<unknown_token>` above is the string `Object`, it means that you're injecting using an type/interface without a proper provider's token. To fix that, make sure you're importing the class reference or use a custom token with `@Inject()` decorator. Read the [custom providers page](/fundamentals/custom-providers).
+
+Also, make sure you didn't end up injecting the provider on itself because self-injections are not allowed on NetJS. When this happens, `<unknown_token>` will likely be equal to `<provider>`.
+
+If you are in a **monorepo setup**, you may face the same error as above but for core provider called `ModuleRef` as a `<unknown_token>`:
+
+```bash
+Nest can't resolve dependencies of the <provider> (?).
+Please make sure that the argument ModuleRef at index [<index>] is available in the <module> context.
+...
+```
+
+This likely happens when your project end up loading two Node modules of the package `@nestjs/core`, like this:
+
+```text
+.
+├── package.json
+├── apps
+│   └── api
+│       └── node_modules
+│           └── @nestjs/bull
+│               └── node_modules
+│                   └── @nestjs/core
+└── node_modules
+    ├── (other packages)
+    └── @nestjs/core
+```
+
+Solutions:
+
+- For **Yarn** Workspaces, use the [nohoist feature](https://classic.yarnpkg.com/blog/2018/02/15/nohoist) to prevent hosting the package `@nestjs/core`.
```

Comment:
```suggestion
- For **Yarn** Workspaces, use the [nohoist feature](https://classic.yarnpkg.com/blog/2018/02/15/nohoist) to prevent hoisting the package `@nestjs/core`.
```